config/routes.rb

```diff
@@ -46,6 +46,10 @@ def matches?(request)
         put  '/:authenticator/:account/api_key'  => 'credentials#rotate_api_key'
 
         post '/authn-k8s/:service_id/inject_client_cert' => 'authenticate#k8s_inject_client_cert'
+
+        constraints authenticator: /|authn-azure|authn-iam|authn-k8s|authn-ldap|authn-gcp|authn-oidc/ do
+          post '/:authenticator(/:service_id)/:account' => 'authenticators#persist_auth'
```

Comment:
Instead of including the `service_id` in the path, let's move it to the body, like this:

```
POST /authn-{authenticator}/{account}}
 {
 "service_id": "<service-id>"
 ...
 }
```
This will give the endpoint a more RESTful convention. For context, when we add them:
```
# Update - future effort
PUT /authn-{authenticator}/{service-id}/{account}}
 {
 ...
 }
# Delete - future effort
DELETE /authn-{authenticator}/{service-id}/{account}}
```

Reply:
Okay, I just moved the service id out of the url and into the body of the request. Should be all set now